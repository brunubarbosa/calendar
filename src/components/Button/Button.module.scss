@import "../../styles/themes";

.button {
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: bold;
  font-size: 14px;
  line-height: 10.8px;
  cursor: pointer;
  color: $light;
  border-radius: 1000px;
  pointer-events: all;
  border-width: 0px;
  border: 1px solid transparent;

  > svg {
    margin-right: 5px;
  }

  &:focus {
    outline: none;
  }

  &.primary {
    color: $salmon;
    border: 1px solid transparent;
    background-color: $primary;
    border: solid 1px rgba(255, 255, 255, 0.16);
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.16);

    &:hover {
      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
    }
  }

  &.accentPrimary {
    color: $white;
    border: 1px solid transparent;
    background-color: $salmon;
    border: solid 1px rgba(255, 255, 255, 0.16);
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.16);

    &:hover {
      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
    }
  }

  &.secondary {
    background-color: transparent;
    color: $salmon;
  }

  &.disabled {
    background-color: $white;
    border-color: $dark;
    color: $dark;
    cursor: not-allowed;
    opacity: 0.6;
  }

  &.small {
    padding: 9.5px 16px;
  }

  &.large {
    padding: 14px 26px;
  }
  &.full {
    width: 100%;
    padding: 15px 26px;
  }
}
