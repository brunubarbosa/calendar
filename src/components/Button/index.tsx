import React from 'react';

import styles from './Button.module.scss';

export type Kind =
  | 'primary'
  | 'secondary'
  | 'accentPrimary'
  | 'accentSecondary'
  | 'negativePrimary'
  | 'negativeSecondary';

export type Size = 'small' | 'large';

export type Type = 'button' | 'submit' | 'reset';

export interface ButtonProps {
  text?: string;
  onClick?: () => void;
  isLoading?: boolean;
  disabled?: boolean;
  kind?: Kind;
  size?: Size;
  type?: Type;
  full?: boolean;
}
const Button: React.FC<ButtonProps> = ({
  text,
  children,
  onClick,
  disabled,
  kind = 'primary',
  size = 'small',
  type = 'button',
  full = false,
}) => {
  return (
    <button
      className={`${styles.button}
        ${styles[size]}
        ${disabled ? styles.disabled : styles[kind]}
        ${full ? `${styles.full}` : ''}
        `}
      type={type}
      onClick={onClick}
      disabled={disabled || undefined}
    >
      {text || children}
    </button>
  );
};

export default Button;
