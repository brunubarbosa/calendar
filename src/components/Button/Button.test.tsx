import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from './index';

const mockfc = jest.fn();

describe('The button', () => {
  it('Should take a snapshot', () => {
    const {container} = render(<Button onClick={mockfc}>Default</Button>);

    expect(container).toMatchSnapshot();
  });
  it('Should renders with full width', () => {
    const {container} = render(
      <Button full onClick={mockfc}>
        Default
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should calls the onCLick callback', () => {
    const {getByText} = render(
      <Button onClick={mockfc}>it is a button</Button>,
    );

    fireEvent.click(getByText('it is a button'));

    expect(mockfc).toHaveBeenCalledTimes(1);
  });

  it('is unclickable when disabled', () => {
    render(
      <Button onClick={mockfc} disabled>
        testButton
      </Button>,
    );

    expect(mockfc).toHaveBeenCalledTimes(0);
  });
});
