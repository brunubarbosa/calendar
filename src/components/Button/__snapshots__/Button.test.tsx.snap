// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The button Should renders with full width 1`] = `
<div>
  <button
    class="button
        small
        primary
        full
        "
    type="button"
  >
    Default
  </button>
</div>
`;

exports[`The button Should take a snapshot 1`] = `
<div>
  <button
    class="button
        small
        primary
        
        "
    type="button"
  >
    Default
  </button>
</div>
`;
